frontend/src/store/summaryDeclaration.js

```diff
@@ -30,6 +30,74 @@ function getProgramYear(selectedGuid, programYearList) {
   return programYear;
 }
 
+/**
+ * Contact the various endpoints to summarize all the details that a client needs to know about any given facility in
+ * their summary declaration.
+ *
+ * @param {Object} facility - The facility to fill with details
+ */
+async function mapFacility(facility) {
+  const applicationStore = useApplicationStore();
+  const appStore = useAppStore();
+  let facilityLicenseResponse = undefined;
+  try {
+    facilityLicenseResponse = (
+      await ApiService.apiAxios.get(`${ApiRoutes.FACILITY}/${facility.facilityId}/licenseCategories`)
+    ).data;
+    facility.licenseCategories = parseLicenseCategories(facilityLicenseResponse);
+  } catch {
+    console.log('error, unable to get childcare category for provider: ', facility.facilityId);
+  }
+
+  // check for opt out - no need for more calls if opt-out
+  if (facility.ccfri?.ccfriId && facility.ccfri?.ccfriOptInStatus == 1) {
+    const ccfriPromises = [ApiService.apiAxios.get(`${ApiRoutes.CCFRIFACILITY}/${facility.ccfri.ccfriId}`)];
+    const afterLoadHooks = [
+      (data) => {
+        facility.ccfri.childCareTypes = data.childCareTypes;
+        facility.ccfri.dates = data.dates;
+      },
+    ];
+
+    // load up the previous ccfri app if it exists, so we can check that we are not missing any child care fee
+    // categories from the last year.
+    if (facility.ccfri.previousCcfriId) {
+      ccfriPromises.push(ApiService.apiAxios.get(`${ApiRoutes.CCFRIFACILITY}/${facility.ccfri.previousCcfriId}`));
+      afterLoadHooks.push((data) => (facility.ccfri.prevYearCcfriApp = data));
+    }
+
+    if (facility.ccfri?.hasRfi || facility.ccfri?.unlockRfi) {
+      ccfriPromises.push(ApiService.apiAxios.get(`${ApiRoutes.APPLICATION_RFI}/${facility.ccfri.ccfriId}/rfi`));
+      afterLoadHooks.push((data) => (facility.rfiApp = data));
+    }
+
+    if (facility.ccfri?.hasNmf || facility.ccfri?.unlockNmf) {
+      ccfriPromises.push(ApiService.apiAxios.get(`${ApiRoutes.APPLICATION_NMF}/${facility.ccfri.ccfriId}/nmf`));
+      afterLoadHooks.push((data) => (facility.nmfApp = data));
+    }
+
+    const APIResponses = await Promise.all(ccfriPromises);
+    const dataFromResponses = APIResponses.map((res) => res.data);
+
+    facility.ccfri.childCareLicenses = facilityLicenseResponse; // jb - so I can build the CCFRI section
+    const ccofProgramYearId = applicationStore.programYearId;
+    const programYearList = appStore.programYearList.list;
+    facility.ccfri.currentYear = getProgramYear(ccofProgramYearId, programYearList);
+    facility.ccfri.prevYear = getProgramYear(facility.ccfri.currentYear.previousYearId, programYearList);
+
+    for (let i = 0; i < afterLoadHooks.length; i++) {
+      const hook = afterLoadHooks[i];
+      const data = dataFromResponses[i];
+      hook(data);
+    }
+  }
+
+  // jb changed below to work with renewel apps
+  facility.facilityInfo = (await ApiService.apiAxios.get(`${ApiRoutes.FACILITY}/${facility.facilityId}`)).data;
```

Comment:
This could totally be another "API call" hook using the above pattern and would help to keep this whole function working with parallel API calls.

Reply:
Resolving as we aren't/shouldn't be doing this on the front-end anymore, anyway.